src/main/java/com/github/mob41/blapi/pkt/CmdPacket.java

```diff
@@ -108,9 +108,19 @@ public CmdPacket(Mac targetMac, int count,
 
         log.debug("Running checksum for headers");
 
+		/* The program is ported from python.
+		 * Byte in python in the range is 0 to 256, but in java, it is in the range of -128 to 127.
+		 * Here is a judgment to ensure that payload[i] is a positive number and checksum is aways growing.         *         * 
+		 *  
+		 * The same problem exists in data[0x20] and data[0x21], but it does not affect the execution of the program.
+		 */
         int checksum = 0xbeaf;
         for (int i = 0; i < payload.length; i++){
-            checksum += payload[i];
+        	if(payload[i]>=0){

```

Comment:
Hmm... I don't think so we should ensure that it is a positive number. In my opinion, if you count on bits, for example:

-1 = 1111 1111
255 = 1111 1111

If you consider a variable which stores 255 into a signed byte in C, and a variable which stores -1 into a unsigned byte in C, the bit they result are the same.

When you add 256 to -1:

  0000 1111 1111
  0001 0000 0000
+_________________
  0001 1111 1111
The excess leading bit may affect further instructions as 255 is not 256.

😰 As I am not good at handling bits, my statement may be wrong. Can you explain more how adding 256 works?